lib/features/login/data/network/dns_lookup/dns_lookup_manager.dart

```diff
@@ -0,0 +1,97 @@
+import 'dart:async';
+
+import 'package:core/utils/app_logger.dart';
+import 'package:core/utils/build_utils.dart';
+import 'package:super_dns_client/super_dns_client.dart';
+import 'package:tmail_ui_user/features/login/data/network/dns_lookup/dns_lookup_priority.dart';
+
+/// Handles DNS SRV lookups for JMAP service discovery.
+///
+/// The manager attempts lookups in order of priority:
+/// **System → Public UDP → Public DoH → Cloud (Google/Cloudflare)**.
+class DnsLookupManager {
+  static const String _jmapServicePrefix = '_jmap._tcp';
+  static const Duration _defaultTimeout = Duration(seconds: 3);
+
+  /// Builds the JMAP SRV hostname from [emailAddress].
+  ///
+  /// Example:
+  /// ```
+  /// input : user@example.com
+  /// output: _jmap._tcp.example.com
+  /// ```
+  String buildJmapHostName(String emailAddress) {
+    final parts = emailAddress.split('@');
+    if (parts.length != 2 || parts[1].isEmpty) {
+      throw ArgumentError('Invalid email address: $emailAddress');
+    }
+    return '$_jmapServicePrefix.${parts[1]}';
+  }
+
+  /// Creates the appropriate [DnsClient] for the given [priority].
+  DnsClient createClient(DnsLookupPriority priority) {
+    const debug = BuildUtils.isDebugMode;
+    switch (priority) {
+      case DnsLookupPriority.system:
+        return SystemUdpSrvClient(debugMode: debug);
+      case DnsLookupPriority.publicUdp:
+        return PublicUdpSrvClient(debugMode: debug);
+      case DnsLookupPriority.publicDoh:
+        return DnsOverHttpsBinaryClient(debugMode: debug);
+      case DnsLookupPriority.cloud:
+        return DnsOverHttps.google(debugMode: debug);
```

Comment:
why only google? how about cloudfare?

Reply:
We are just creating class `DnsOverHttps` and in class we will have method to handle both google and cloudfare

Reply:
Use `DnsOverHttps.empty()` for clarity